shipctl/x86_64/WindowsServer_2016/utility.ps1

```diff
@@ -283,6 +283,122 @@ function copy_file_from_prev_state([string] $filePath, [string] $restorePath) {
     Copy-Item -Recurse -Verbose "$previousStateFile" "$restorePath"
 }
 
+function replicate() {
+
+    param(
+    [string]$resourceFrom,
+    [string]$resourceTo,
+    [switch]$files_only,
+    [switch]$metadata_only,
+    [switch]$webhook_data_only
+  )
+  write-output "here we go"
+  if ($PSBoundParameters.Count -lt 2) {
+    throw "Usage: shipctl replicate FROM_resource TO_resource [-files_only|-metadata_only|-webhook_data_only]"
+  }
+  $toCurrentJob = $false
+  $typeFrom = $(get_resource_type "$resourceFrom")
+  $typeTo = $(get_resource_type "$resourceTo")
+
+  if ($resourceFrom -eq $env:JOB_NAME) {
+    throw "Error: current job cannot be the FROM argument"
+  }
+  if ($typeTo -eq "ciRepo") {
+    throw "Error: cannot replicate to ciRepo"
+  }
+  if (($typeFrom -match "^gitRepo|ciRepo|syncRepo$") -and ($typeTo -match "^gitRepo|syncRepo$")) {
+    $metadata_only = $true
+  } elseif ("$resourceTo" -eq "$env:JOB_NAME") {
+    $toCurrentJob = $true
+  } elseif ("$typeFrom" -ne "$typeTo") {
+    throw "Error: resources must be the same type"
+  }
+
+  # copy files
+  if (!$metadata_only) {
+    $pathFrom = "$env:JOB_PATH/IN/$resourceFrom/$typeFrom"
```

Comment:
yes, however I decided to manually add the paths because of how state resources are handled when they are both IN and OUT.  get_resource_state relies on ENVs, and since the resource name is the same, the OUT ENVs override the IN envs, so you can never get the IN/from path.